### PR TITLE
AF-2813: java.io.IOException: Too many open files using business-central

### PR DESCRIPTION
**Pull request by akumar074**
* Deleting all branches index when deleting a project

(cherry picked from commit a2e6ee6b2c0867c9389876b7aa799a74b520bb7c)

**JIRA**:
[RHPAM-3309](https://issues.redhat.com/browse/RHPAM-3309)
[AF-2813](https://issues.redhat.com/browse/AF-2813)

**referenced Pull Requests**: 
Original PR : https://github.com/kiegroup/appformer/pull/1108

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by akumar074** [comment]
Jenkins run fdb

**Review by adrielparedes — APPROVED** [review]

**Comment by akumar074** [comment]
Jenkins run cdb

**Comment by akumar074** [comment]
Jenkins run cdb

**Review by barboras7 — APPROVED** [review]

**Comment by akumar074** [comment]
Jenkins run cdb

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-lucene/src/test/java/org/uberfire/ext/metadata/backend/lucene/index/LuceneInedexManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.uberfire.ext.metadata.model.KCluster;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LuceneInedexManagerTest {
```
Comment (ederign) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-lucene/src/test/java/org/uberfire/ext/metadata/backend/lucene/index/LuceneInedexManagerTest.java:15]: @akumar074 in another PR (could be whenever you do another change), could you please fix the name of this class?

**Pull request merged**